Amazon.Ion.ObjectMapper/IonObjectSerializer.cs

```diff
@@ -31,9 +31,36 @@ public override object Deserialize(IIonReader reader)
             IonType ionType;
             while ((ionType = reader.MoveNext()) != IonType.None)
             {
-                var property = FindProperty(reader.CurrentFieldName);
+                MethodInfo method;
+                PropertyInfo property;
                 FieldInfo field;
-                if (property != null)
+
+                // Check if current Ion field has a IonPropertySetter annotated setter method.
+                if ((method = FindSetter(reader.CurrentFieldName)) != null)
+                {
+                    // A setter should be a void method.
+                    if (method.ReturnParameter?.ParameterType != typeof(void))
+                    {
+                        continue;
```

Comment:
I don't think we care if the setter returns anything, just ignore this

Reply:
Addressed.